File: task4.py
def original_program(A, B):
    A = A - B
    C = A * 2
    return C

def incorrect_programs(A, B):
    incorrect_results = []
    # For incorrect "-" operator
    incorrect_results.append((A + B) * 2)
    incorrect_results.append((A * B) * 2)
    incorrect_results.append((A / B) * 2)
    # For incorrect "*" operator
    incorrect_results.append(A - B + 2)
    incorrect_results.append(A - B - 2)
    incorrect_results.append((A - B) / 2)
    # For both incorrect
    incorrect_results.extend([(A + B) + 2, (A * B) + 2, (A / B) + 2])
    incorrect_results.extend([(A + B) - 2, (A * B) - 2, (A / B) - 2])
    incorrect_results.extend([(A + B) / 2, (A * B) / 2, (A / B) / 2])
    
    return incorrect_results

B = 1
ambiguous_A_values = []
for A in range(-100, 101):  # arbitrary range from -100 to 100, can be expanded
    orig_result = original_program(A, B)
    for incorrect_result in incorrect_programs(A, B):
        if orig_result == incorrect_result:
            ambiguous_A_values.append(A)
            break  # No need to check other incorrect programs for this A value

print("Ambiguous A values for B=1 are:", ambiguous_A_values)

File: task4.java
import java.util.ArrayList;
import java.util.List;

public class TestProgram {

    public static double originalProgram(double A, double B) {
        A = A - B;
        double C = A * 2;
        return C;
    }

    public static List<Double> incorrectPrograms(double A, double B) {
        List<Double> incorrectResults = new ArrayList<>();
        
        // For incorrect "-" operator
        incorrectResults.add((A + B) * 2);
        incorrectResults.add((A * B) * 2);
        incorrectResults.add((A / B) * 2);

        // For incorrect "*" operator
        incorrectResults.add(A - B + 2);
        incorrectResults.add(A - B - 2);
        incorrectResults.add((A - B) / 2);

        // For both incorrect
        incorrectResults.add((A + B) + 2);
        incorrectResults.add((A * B) + 2);
        incorrectResults.add((A / B) + 2);
        incorrectResults.add((A + B) - 2);
        incorrectResults.add((A * B) - 2);
        incorrectResults.add((A / B) - 2);
        incorrectResults.add((A + B) / 2);
        incorrectResults.add((A * B) / 2);
        incorrectResults.add((A / B) / 2);
        
        return incorrectResults;
    }

    public static void main(String[] args) {
        double B = 1;
        List<Double> correctValues = new ArrayList<>();

        for (double A = -10; A <= 10; A++) {
            double origResult = originalProgram(A, B);
            boolean isCorrectValues = true;
            
            for (double incorrectResult : incorrectPrograms(A, B)) {
                if (origResult == incorrectResult) {
                    isCorrectValues = false;
                    break;  // This A value gives the same result in incorrect program
                }
            }
            
            if (isCorrectValues) {
                correctValues.add(A);
            }
        }

        System.out.println("Correct test A values for B=1 are: " + correctValues);
    }
}
